src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,37 @@
+package duke.tasks;
+
+public class Deadline extends Task {
+
+    private final String by;
+
+    /**
+     * Initialises a Deadline task.
+     *
+     * @param description The deadline description.
+     * @param by          The time at which the deadline is up.
+     */
+    public Deadline(String description, String by) {
```

Comment:
Using `String` to store the actual deadline is okay to me, but for greater consistency, you may also consider storing it as `Date` objects. When you export the tasks to a file, the deadlines could be converted from `Date` to `String` and hence follow the same format.